client/app/components/Parameters.jsx

```diff
@@ -140,7 +142,22 @@ export class Parameters extends React.Component {
     const { queryResultErrorData } = this.props;
     const error = get(queryResultErrorData, ['parameters', param.name], false);
     if (error) {
-      return [error, 'error'];
+      const feedback = <Tooltip title={error}>{error}</Tooltip>;
+      return [feedback, 'error'];
+    }
+
+    // unsaved
+    const { unsavedParameters } = this.props;
+    if (includes(unsavedParameters, param.name)) {
+      const feedback = (
+        <>
+          Unsaved{' '}
+          <Tooltip title='Click the "Save" button to preserve this parameter.'>
```

Comment:
@susodapop any better suggestions for this text?

Reply:
'Click the "Save" button to make this the default value for this parameter.'

Reply:
But it's actually for the parameter itself - not its value.
The case is when a new parameter is typed into the query but ain't saved to DB till the user hits the "save" button.

Reply:
@ranbena Forgot to tag you in the previous reply.

Reply:
In that case it's no different than any other change to a query. Is a special indication necessary?

Reply:
It's not a must, but the asterisk indication imo is not clear enough for changes that are represented below it (including parameter setting changes which is out of scope).
I encourage you to experience it and see for your self if it provides a better user experience.

Reply:
I agree the asterisk is sub-optimal. Could we replace it with an indicator that is more obvious?

If you keep the tooltip then the text you've written above is fine.

Reply:
One way to go about it is not show or disable the Save button when query isn't dirty.
So when it is, there would be a clear indication that there are changes to be saved.

Another reason I wanted this inline indication was so that it would be clear which parameter corresponds to the one added. So "Unsaved" could be substituted with "New parameter" with a green color instead.

@arikfr how do you feel about this feature?